docs/reference/bare_metal/c_small_libs.md

```diff
@@ -0,0 +1,55 @@
+<h1 id="using-small-c-libraries">Using small C libraries in Mbed OS</h1>
+
+If your application does not use a RTOS then you should build it in the bare metal mode to achieve memory savings. Both the `ARM` and `GCC_ARM` toolchains support code optimised versions of their C standard libraries, `microlib` and `newlib-nano`. It is safe to use these smaller C libraries in bare metal mode and we recommend using them. Note that a Mbed OS application can build with these libraries in non bare metal mode but you need to be aware of some restrictions in their usage, in particular in regard to thread-safety, as will be described later.
```

Comment:
> Note that a Mbed OS application can build with these libraries in non bare metal mode but you need to be aware of some restrictions in their usage, in particular in regard to thread-safety, as will be described later.

I wouldn't mention that. I would also go a step further and disallowed RTOS small lib builds. Mbed OS and RTX have their own threads, so even if user doesn't create any threads the whole thing can explode on its own.

Reply:
There is nothing in the build system that prevents using small lib with a RTOS. The purpose of this documentation is to describe what we have.
It is emphasized that in bare metal mode small library should be used and that a RTOS application should not use it. I can explicitly say that it is not recommended or I can drop that sentence.

Reply:
Sentence removed in latest commit.